pkg/ctrlserver/graph/schema.resolvers.go

```diff
@@ -707,16 +707,16 @@ func (r *podResolver) Logs(ctx context.Context, obj *v1.Pod) (string, error) {
 }
 
 func (r *podResolver) Daemon(ctx context.Context, obj *v1.Pod) (*v1.Pod, error) {
-	var list v1.PodList
-	if err := r.Client.List(ctx, &list, client.MatchingLabels(componentLabels(model.ComponentDaemon))); err != nil {
+	daemons, err := getDaemonMap(ctx, r.Client)
+	if err != nil {
 		return nil, err
 	}
-	for _, daemon := range list.Items {
-		if obj.Spec.NodeName == daemon.Spec.NodeName {
-			return &daemon, nil
-		}
+
+	daemon, exist := daemons[obj.Name]
+	if !exist {
+		return nil, fmt.Errorf("daemon of pod(%s/%s) not found", obj.Namespace, obj.Name)
```

Comment:
s/fmt.Errorf/errors.Errorf/g?

Reply:
Currently, we cannot use github.com/golang/errors in schema.resolvers.go because a bug of gqlgen (it will replace github.com/golang/errors with errors in stdlib). I will try to upgrade the version of gqlgen in another PR.